localstack-core/localstack/services/lambda_/invocation/execution_environment.py

```diff
@@ -402,3 +403,11 @@ def _get_execution_timeout_seconds(self) -> int:
         if is_lambda_debug_timeout_for(self.function_version.qualified_arn):
             return DEFAULT_LAMBDA_DEBUG_MODE_TIMEOUT_SECONDS
         return self.function_version.config.timeout
+
+    def _get_startup_timeout_seconds(self) -> int:
+        # Returns the timeout value in seconds to be enforced during lambda container startups.
+        # This is the value defined through LAMBDA_RUNTIME_ENVIRONMENT_TIMEOUT or the LAMBDA
+        # DEBUG MODE default if this is enabled.
+        if is_lambda_debug_timeout_for(self.function_version.qualified_arn):
```

Comment:
I think we should create a follow-up docs PR and list the implications of the Lambda debug mode.
We could also add a debug log here to make the config adjustment traceable at runtime (similar for other potentially implicit config updates) 🤔 Do you think logging is helpful or verbose clutter @dfangl @dominikschubert?

Reply:
I think it's a nice idea to see those changes live. However, currently users are using debug mode very often, which might lead to being too verbose. As long as the config changes are limited to a handful, I think it is fine, once it gets too much, maybe one consolidated line of changes?